NimbleMedium/Sources/Presentation/Views/Screen.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Screen.swift
+//  NimbleMedium
+//
+//  Created by Mark G on 10/08/2021.
+//
+
+import SwiftUI
+
+struct Screen<Content>: View where Content: View {
+
+    private let content: Content
```

Comment:
I would prefer to define the property like the code below as we usually create and use the view in `body`
```suggestion
private let content: () -> Content
```

In some cases, we need to duplicate the content ([GridStack](https://www.hackingwithswift.com/books/ios-swiftui/custom-containers)), so don't need to create the view in init

Reply:
@suho  Hmm I think we should use it correctly, for example
```swift
struct HomeView: View {

    private let content: () -> Content
    @State var buttonTitle: String = "Append me"

    var body: some View {
        print("Rebuild")

        // TODO: Update the title and its content in Home integrate task
        return Screen(title: Localizable.homeTitle()) {
            content()
            Button {
                buttonTitle += "1"
            } label: {
                Text(buttonTitle)
            }

        }
    }
}
```
In this case, every time we click the button, the body will be recalled. Then if we use `content()` here, it will be re-created also

Reply:
I think It's normal.
From this link: https://www.hackingwithswift.com/books/ios-swiftui/custom-containers. It defines `let content: (Int, Int) -> Content`.

also, with this code, another views will be re-created, right?
```swift
var body: some View {
       Screen(title: Localizable.homeTitle()) {
            content()
            Button {
                buttonTitle += "1"
            } label: {
                Text(buttonTitle) // <--- everytime we press the button, it will re-created the Text?
            }
            Text("Another Text") // <--- it will re-created the Text?
            HStack {} // <--- it will re-created the HStack?
        }
    }
```

Correct me if I'm missing something :D

Reply:
@markgravity However, I think we can go with both ways. In some cases like the GridStack in the article, we can use `let content: (Int, Int) -> Content`, otherwise, we can use `let content: Content`
Not a big deal here :D